kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/configuration/NMConfigurationServiceImpl.java

```diff
@@ -383,8 +390,7 @@ private boolean isNatValid(String interfaceName) {
         Optional<NetInterfaceStatus> status = getNetInterfaceStatus(interfaceName);
 
         if (type.isPresent() && isNatEnabled.isPresent() && status.isPresent()) {
-            boolean isSupportedType = type.get() == NetInterfaceType.ETHERNET || type.get() == NetInterfaceType.WIFI
-                    || type.get() == NetInterfaceType.MODEM;
+            boolean isSupportedType = SUPPORTED_NAT_INTERFACE_TYPES.contains(status);
```

Comment:
Shouldn't this be `type`?

```suggestion
            boolean isSupportedType = SUPPORTED_NAT_INTERFACE_TYPES.contains(type.get());
```